### PR TITLE
Add devise gem and User model

### PR DESCRIPTION
**Pull request by storresy**
**Summary**
- Added Devise gem
- Generated User model using devise generator
- Added fields first_name and last_name to User model

**Trello Card**
https://trello.com/c/dWcHFIDY

**Screenshots**
screen shot 2018-12-07 at 15 03 32

### CODE REVIEW

File: db/migrate/20181207191632_devise_create_users.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
```
Comment (lmasello) [inline db/migrate/20181207191632_devise_create_users.rb:1]: We can avoid this by adding the following statement to the .rubocop.yml file:
```
Style/FrozenStringLiteralComment:
  Enabled: false
```

**Review by lmasello — APPROVED** [review]

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (Gmarn) [inline app/models/user.rb:5]: Becarefull ir you add some options you do not need. As `recoverable`, `rememberable`

File: db/migrate/20181207191632_devise_create_users.rb (hunk 1 above)
Comment (Gmarn) [reply db/migrate/20181207191632_devise_create_users.rb:1]: Also rubocop should not pass on migration file.

File: config/initializers/devise.rb [hunk 3]
```diff
@@ -0,0 +1,290 @@
+# frozen_string_literal: true
```
Comment (Gmarn) [inline config/initializers/devise.rb:1]: I do not think you need that

File: db/migrate/20181207191632_devise_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
+
+      t.string :first_name, null: false
+      t.string :last_name, null: false
```
Comment (Gmarn) [inline db/migrate/20181207191632_devise_create_users.rb:36]: 👏

**Pull request merged**